pandas/core/groupby/groupby.py

```diff
@@ -617,44 +618,53 @@ def _make_wrapper(self, name):
         def wrapper(*args, **kwargs):
             # a little trickery for aggregation functions that need an axis
             # argument
-            kwargs_with_axis = kwargs.copy()
-            if "axis" not in kwargs_with_axis or kwargs_with_axis["axis"] is None:
-                kwargs_with_axis["axis"] = self.axis
-
-            def curried_with_axis(x):
-                return f(x, *args, **kwargs_with_axis)
+            kwargs2 = kwargs.copy()
 
             def curried(x):
-                return f(x, *args, **kwargs)
+                return f(x, *args, **kwargs2)
 
             # preserve the name so we can detect it when calling plot methods,
             # to avoid duplicates
-            curried.__name__ = curried_with_axis.__name__ = name
+            curried.__name__ = name
 
             # special case otherwise extra plots are created when catching the
             # exception below
             if name in base.plotting_methods:
                 return self.apply(curried)
 
-            try:
-                return self.apply(curried_with_axis)
-            except Exception:
-                try:
-                    return self.apply(curried)
-                except Exception:
+            import inspect
 
-                    # related to : GH3688
-                    # try item-by-item
-                    # this can be called recursively, so need to raise
-                    # ValueError
-                    # if we don't have this method to indicated to aggregate to
-                    # mark this column as an error
-                    try:
-                        return self._aggregate_item_by_item(name, *args, **kwargs)
-                    except AttributeError:
-                        # e.g. SparseArray has no flags attr
-                        raise ValueError
+            sig = inspect.signature(f)
```

Comment:
Could this be done before the currying? IIUC correctly we are mutating the kwargs after the function has been defined but before it gets executed. I think would be more intuitive if we set up the appropriate kwargs before the currying even happens

Reply:
i wasnt sure what the semantics are of editing kwargs.  is it necessarily a new dict, or could we be editing a dict that exists in the calling namespace?

Reply:
I don't know definitely but I think the same rules would apply to kwargs as to default arguments, i.e. the parameter values get evaluated on function execution and not definition

https://docs.python.org/3/reference/compound_stmts.html#function-definitions

Reply:
If there wasn't the `if name in base.plotting_methods` case just above this I would think the answer is "yes" we can do this earlier, but I'm not confident as-is.  Good candidate for follow-up to see if we can get that case out of this wrapper.